apps/condo/domains/marketplace/components/Invoice/BaseInvoiceForm.tsx

```diff
@@ -0,0 +1,1095 @@
+                                                        rows: {
+                                                            ...form.getFieldValue('rows'),
+                                                            [marketItemForm.name]: {
+                                                                ...form.getFieldValue(['rows', marketItemForm.name]),
+                                                                toPay: option.isMin ? `${FromMessage} ${option.toPay}` : option.toPay,
+                                                                isMin: option.isMin,
+                                                            },
+                                                        },
+                                                    })
+
+                                                    form.validateFields([['rows', marketItemForm.name, 'toPay']])
+                                                }}
+                                                onChange={text => {
+                                                    const existedMarketItem = flatMarketOptions.find(marketItem => marketItem.label === text)
+                                                    const sku = existedMarketItem ? existedMarketItem.sku : null
+
+                                                    form.setFieldsValue({
+                                                        rows: {
+                                                            ...form.getFieldValue('rows'),
+                                                            [marketItemForm.name]: {
+                                                                ...form.getFieldValue(['rows', marketItemForm.name]),
+                                                                sku,
+                                                            },
+                                                        },
+                                                    })
+                                                }}
+                                            />
+                                        </Form.Item>
+                                    </Col>
+                                    <Col xs={24} lg={4}>
+                                        <Form.Item
+                                            label={QuntityLabel}
+                                            name={[marketItemForm.name, 'count']}
+                                            required
+                                            labelAlign='left'
+                                            labelCol={{ span: 24 }}
+                                            rules={[requiredValidator]}
+                                            initialValue={1}
+                                        >
+                                            <Select
+                                                disabled={disabled}
+                                                options={[...Array(50).keys() ].map( i => ({
```

Comment:
It's not memoized. It seems that with each rerender this array and label, key will be created. Perhaps it’s worth removing this from the layout into the component itself and memoizing it.
ps: minor, does not affect approval